external-crates/move/crates/move-analyzer/trace-adapter/src/adapter.ts

```diff
@@ -268,41 +272,92 @@ export class MoveDebugSession extends LoggingDebugSession {
         this.sendResponse(response);
     }
 
+    /**
+     * Converts a runtime value to a DAP variable.
+     *
+     * @param value variable value
+     * @param name variable name
+     * @param type optional variable type
+     * @returns a DAP variable.
+     */
+    private convertRuntimeValue(
+        value: RuntimeValueType,
+        name: string,
+        type?: string
+    ): DebugProtocol.Variable {
+        if (typeof value === 'string') {
+            return {
+                name,
+                type,
+                value,
+                variablesReference: 0
+            };
+        } else if (Array.isArray(value)) {
+            const compoundValueReference = this.variableHandles.create(value);
+            return {
+                name,
+                type,
+                value: '(' + value.length + ')[...]',
+                variablesReference: compoundValueReference
+            };
+        } else {
+            const compoundValueReference = this.variableHandles.create(value);
+            const accessChainParts = value.type.split('::');
+            const datatypeName = accessChainParts[accessChainParts.length - 1];
```

Comment:
Wondering if the `datatypeName` should include the module name as well? Although by that logic we should include the address as well and display the fully qualified typename which isn't really practical...

Not needed now, but more something to think about for the future probably is if there is any way to provide a "mouse-over" name here (or a expandable name) where you could get the fully-qualified typename of the type if you want?

Reply:
It'a a great idea actually! I guess, I did not think about it enough... In any case, I added support for on-hover showing the fully qualified type for non-primitive values. Here are some examples:
![image](https://github.com/user-attachments/assets/10f60fe5-9d4b-4376-acf9-a52119a62e3e)

![image](https://github.com/user-attachments/assets/67bfe5f4-2a5d-4111-a058-d7aed2c475c5)

![image](https://github.com/user-attachments/assets/8c67b392-d449-4150-8232-60e94ca09a86)

I also added support for the same info being displayed for inner fields of structs/enums:
![image](https://github.com/user-attachments/assets/768d83f0-42e5-40ca-804b-0596b18cf59c)

Reply:
Amazing! That's awesome :)